hoodie-client/src/main/java/com/uber/hoodie/io/HoodieUpdateHandle.java

```diff
@@ -133,16 +135,27 @@ private void writeUpdateRecord(HoodieRecord<T> hoodieRecord, IndexedRecord index
     public void write(GenericRecord oldRecord) {
         String key = oldRecord.get(HoodieRecord.RECORD_KEY_METADATA_FIELD).toString();
         HoodieRecord<T> hoodieRecord = keyToNewRecords.get(key);
+        boolean copyOldRecord = true;
         if (keyToNewRecords.containsKey(key)) {
             try {
                 IndexedRecord avroRecord = hoodieRecord.getData().combineAndGetUpdateValue(oldRecord, schema);
-                writeUpdateRecord(hoodieRecord, avroRecord);
+                if (writeUpdateRecord(hoodieRecord, avroRecord)) {
+                    /* ONLY WHEN
+                     * 1) we have an update for this key AND
+                     * 2) We are able to successfully write the the combined new value
+                     *
+                     * We no longer need to copy the old record over.
+                     */
+                    copyOldRecord = false;
```

Comment:
We should probably have a unit test for this. We can may be shelf that as a seperate jira and move on.